doc/DEVELOPMENT.rst

```diff
@@ -0,0 +1,266 @@
+
+To set OpenGL ES 3.0 as the *default* renderer, do:
+
+::
+
+    meson configure -Dr_default=gles30
+
+The OpenGL ES 2.0 backend can be enabled similarly, using ``gles20`` instead of
+``gles30``. However, it requires a few extensions to function correctly, most
+notably:
+
+- ``OES_depth_texture`` or ``GL_ANGLE_depth_texture``
+- ``OES_standard_derivatives``
+- ``OES_vertex_array_object``
+- ``EXT_frag_depth``
+- ``EXT_instanced_arrays`` or ``ANGLE_instanced_arrays`` or
+  ``NV_instanced_arrays``
+
+For Windows and macOS, you will need Google's ANGLE library for both ES 3.0 and
+2.0. You'll need to check out
+`ANGLE <https://github.com/google/angle>`__ and build it first. Refer to their
+documentation on how to do that.
+
+Once you've compiled ANGLE, enable it with:
+
+::
+
+    meson -Dinstall_angle=true -Dangle_libegl=/path/to/libEGL.{dll,dylib}
+    -Dangle_libgles=/path/to/libGLESv2.{dll,dylib}
+
+Ensure you use the correct file extension for your platform. (``.dll`` for
+Windows, ``.dylib`` for macOS.)
+
+It'll install automatically with ``ninja install`` (as mentioned above).
+
+Development
+-----------
+
+Build Options
+^^^^^^^^^^^^^
+
+You can enable debugging options/output for development purposes:
+
+::
+
+    meson configure -Dbuildtype=debug -Db_ndebug=false
+
+
+This option also helps for speeding up build times:
+
+::
+
+    meson configure -Db_lto=false
+
+
+For debugging actual gameplay, you can set this option and it will enable cheats
+and other 'fast-forward' options by the pressing keys defined in
+``src/config.h``.
+
+::
+
+    meson configure -Ddeveloper=true
+
+
+Coding Style
+^^^^^^^^^^^^
+
+In the ``*.c`` files, tabs are used. In the ``meson.build`` and ``*.py`` files, spaces
+are used.
+
+To help you abide by this standard, you should install
+`EditorConfig <https://github.com/editorconfig>`__ for your preferred editor of
+choice, and load in the file found at ``.editorconfig`` in the root of the
+project.
+
+
+Platform-Specific Tips
+^^^^^^^^^^^^^^^^^^^^^^
+
+Windows
+"""""""
+
+TODO: complete this section.
+
+Linux (Debian/Ubuntu)
+"""""""""""""""""""""
+
+On an apt-based system (Debian/Ubuntu), ensure you have build dependencies
+installed:
+
+::
+
+    apt-get install meson cmake build-essential libsdl2-dev libsdl2-mixer-dev libogg-dev libopusfile-dev libpng-dev libzip-dev libx11-dev libwayland-dev
+
+
+macOS
+"""""
+
+Manual
+~~~~~
+
+On macOS, you need to begin with installing the Xcode Command Line Tools:
+
+::
+
+    xcode-select --install
+
+There are additional command line tools that you'll need. You can acquire those
+by using `Homebrew <https://brew.sh/>`__.
+
+Follow the instructions for installing Homebrew, and then install the following
+tools:
+
+::
+
+    brew install meson cmake pkg-config docutils imagemagick pygments
+
+
+The following dependencies are technically optional, and can be pulled in at
+build-time, but you're better off installing them yourself to reduce compile
+times:
```

Comment:
It's probably worth mentioning that it's possible to force internal dependencies with `--wrapmode=forcefallback` when making the build directory. This is used when building the official releases for Windows, macOS, and Emscripten, for consistency and ease of dependency management on my end.